crates/rpc/rpc/src/debug.rs

```diff
@@ -306,7 +306,19 @@ where
                         return Ok(frame.into())
                     }
                     GethDebugBuiltInTracerType::PreStateTracer => {
-                        Err(EthApiError::Unsupported("pre state tracer currently unsupported."))
+                        let prestate_config = tracer_config
+                            .into_pre_state_config()
+                            .map_err(|_| EthApiError::InvalidTracerConfig)?;
+                        let mut inspector = TracingInspector::new(TracingInspectorConfig::from_geth_config(&config));
+
+                        let (res, _, db) = self
+                            .inner
+                            .eth_api
+                            .inspect_call_at_and_return_state(call, at, overrides, &mut inspector)
```

Comment:
just making note here that this will change after #3914

but no action required now